src/SingleCellAnnotator.imjoy.html

```diff
@@ -627,40 +611,47 @@
       if(!sendData.discarded){
         if(this.current_image_data.location_changed){
           sendData.location = {};
-          sendData.location = this.current_image_data.location;
-          let noLocation = true;
-          for (loc in sendData.location){
-            if(sendData.location[loc] === null){
-              sendData.location[loc] = false
-            }
-            else{
-              noLocation = false;
-            }
-          }
-          if (noLocation && this.current_image_data.label_display_mode==='multiple'){
-            const anwser = await api.confirm("No location label selected, is that correct?")
-            if (!anwser){
-              return
+          if (this.current_image_data.location){
+            sendData.location = this.current_image_data.location;
+            let noLocation = true;
+            for (loc in sendData.location){
+              if(sendData.location[loc] === null){
+                sendData.location[loc] = false
+              }
+              else{
+                noLocation = false;
+              }
             }
+            if (noLocation && this.current_image_data.label_display_mode==='multiple'){
+              const anwser = await api.confirm("No location label selected, is that correct?")
+              if (!anwser){
+                return
+              }
+            };
           };
-        }
-        
-  
+        };
         if (this.current_image_data.mask_adjusted){
           sendData.extend = this.current_image_data.extend;
           sendData.bbox = this.current_image_data.bbox;
           sendData.mask = this.current_image_data.mask_geojson;
           // now adjust the digit to 1 decimal for cooridinates
-          let coordinates = this.current_image_data.mask_geojson.features[0].geometry.coordinates[0]
-          for (let index = 0; index < coordinates.length; index++){
-            coordinates[index][0] = Number(coordinates[index][0].toFixed(1));
-            coordinates[index][1] = Number(coordinates[index][1].toFixed(1));
+          let features = this.current_image_data.mask_geojson.features
+          api.alert(JSON.stringify(features))
+          for (let i = 0; i < features.length; i++){
+            features[i].properties ={'label': 'cell'}; // keep the label as cell
+            let coordinates = features[i].geometry.coordinates[0]
+            for (let index = 0; index < coordinates.length; index++){
+              coordinates[index][0] = Number(coordinates[index][0].toFixed(1));
```

Comment:
I think we don't need this. When we call `writeFeaturesObject`, we already set the `decimals=1`.

https://openlayers.org/en/latest/apidoc/module-ol_format_GeoJSON-GeoJSON.html#writeFeaturesObject

Reply:
Now it is removed